.eslintrc

```diff
@@ -9,7 +15,26 @@
     "no-prototype-builtins": "off",
 
     // Replaced by TypeScript's static checking.
-    "react/prop-types": "off"
+    "react/prop-types": "off",
+
+    // Upgrade TS rules from warning to error.
+    "@typescript-eslint/no-unused-vars": "error",
+
+    // Disable TS rules that we dislike.
+    "@typescript-eslint/ban-ts-comment": "off",
+    "@typescript-eslint/no-empty-function": "off",
+    "@typescript-eslint/no-explicit-any": "off",
+    "@typescript-eslint/no-this-alias": "off",
+
+    // Enforce consistency in cases where TypeScript supports old and new
+    // syntaxes for the same thing.
+    //
+    // - Require `<var> as <type>` for casts
+    // - Require `import type` for type imports. The corresponding rule for
+    //   exports is not enabled yet because that requires setting up type-aware
+    //   linting.
+    "@typescript-eslint/consistent-type-assertions": "error",
+    "@typescript-eslint/consistent-type-imports": "error"
```

Comment:
Some or all of these lint changes can eventually be folded into eslint-config-hypothesis.